src/Coupon.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+var Coupon = function(parameters) {
+    this.Code = parameters.code;
+    this.Qty = parameters.quantity ? parameters.quantity : 1;
+    this.ID = 1;
+    this.isNew = true;
+}
```

Comment:
Just a note, this sort of Class is often best built with [Object.defineProperties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperties).

It allows more extensiblity as well as immutable things. In this case it's not completely necessary, but good practice. Allowing public private and protected scopes.